### DISCUSSION TITLE
How to return Multipart data (e.g excel file or photos) using Actix?

### DISCUSSION THREAD

**Discussion by stanley355** [actix/actix-web#2672] [Q&A]
Hi, I've successfully parsed incoming request of excel data using actix and actix multipart, however I'm confused on how to return the excel data as a response, anybody having similar issue???

**Comment by robjtede** [comment]
```rust
HttpResponse::Ok()
  .content_type("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
  .body(excel_bytes)
```

**Reply by stanley355** [reply]
@robjtede could u explain more of how to turn the excel file into bytes??? Sry, new to programming

**Reply by asonix** [reply]
where is the data coming from? did you generate it? is it a file?

**Reply by stanley355** [reply]
Yeah it's an excel file @asonix 

```

pub async fn upload(payload: Multipart) -> Result<HttpResponse, MultipartError> {
  let upload_status = file_controller::ExcelFile::process_file(payload).await;
  
  match upload_status {
    Ok(filename) => Ok(
      HttpResponse::Ok()
        .header(
          "Content-type",
          "application/vnd.ms-excel",
        )
        .body(""),
    ),
    Err(e) => {
      println!("Error: {}", e);
      Ok(HttpResponse::BadRequest().body("upload_failed"))
    }
  }
}
```

The upload status will return `Result<String>` where the string is the path to the excel file, how can I return the body with the excel file converted to bytes??

**Reply by robjtede** [reply]
Ah, if you're serving a file from disk, this is the way to go: https://docs.rs/actix-files/latest/actix_files/struct.NamedFile.html
